### DISCUSSION TITLE
clock_wait()

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#2293] [Q&A] [answered]
Is it possible to call `clock_wait()` when compiling for Cooja?

Is there a way to implement a few millisec blocking delay otherwise? For a scientific experiment I need a function to hold in place as if it was blocking for a very short time).

**Comment by alexrayne** [comment, marked as answer]
no way, do not call it. current cooja have no implementation compatipbe with JNI-comm core. In other platforms it have default implementation as blocking mcu wait.

**Reply by rahav** [reply]
Thank you Alex

**Reply by rajeev1986** [reply]
will this help?

```
#define BUSYWAIT_UNTIL(max_time) \
  do { \
    rtimer_clock_t t0; \
    t0 = RTIMER_NOW(); \
    while(RTIMER_CLOCK_LT(RTIMER_NOW(), t0 + (max_time))) { \
      watchdog_periodic(); \
    } \
  } while(0)
```
  Instead of `clock_wait()`, one can define the `BUSYWAIT_UNTIL()` and use it for instance: 
  
  `BUSYWAIT_UNTIL(RTIMER_SECOND/10)`
  
Correct me if I am wrong, @alexrayne !

**Comment by alexrayne** [comment]
No it s not help. Helps, if you made fullfeatured etimer, and waits it by
PROCESS_WAIT

вт, 14 февр. 2023 г., 01:50 Rajeev Piyare ***@***.***>:

**Comment by alexrayne** [comment]
Guess your simulation may still works, if you waits a few ms, realy short
time.

вт, 14 февр. 2023 г., 01:50 Rajeev Piyare ***@***.***>:
